### ISSUE TITLE
Use a tmp datadir for tests

### ISSUE THREAD

**Issue by carlosms** [src-d/engine#369]
Current tests use `~/.src/gitbsae` as datadir, the same dir used by a normal installation.
This should ideally be a tmp dir, as we do for the workdir.

**Comment by smacker** [comment]
Do you have any ideas on how to approach this?
From the top of my head:
- add an optional argument to `init` command
- add this into config file

**Comment by carlosms** [comment]
Both make sense. We could eventually even allow both options.
Maybe for now a flag for `init` is more similar to how we set the `workdir`, so I would go with that one first.

**Comment by smacker** [comment]
Hm. There is a problem with this approach: we can read flag from cli and pass it to a new instance of the daemon. So the directory is saved inside daemon container. But we also have `prune` command. That doesn't require running daemon.

As the solution, I would propose to save arguments that we passed to daemon into `~/.srcd`. It should also help [here](https://github.com/src-d/engine/blob/350af2d517e0b1d5f6e0be0c94f403474d58e293/cmd/srcd/cmd/errors.go#L19) and maybe in some other cases.

WDYT?

**Comment by smacker** [comment]
and we also have "ensureStarted" that should use the dirs from the previous run but now it is using `pwd`.

**Comment by smacker** [comment]
We agreed to change index directory to docker volumes.
Then we might don't need this flag. I'm pausing working on this.
But the discovered bug about loosing last config will be fixed in separate issue.

**Comment by carlosms** [comment]
After #373 index data is already stored in volumes, so I think this issue is not needed anymore.
